fv3config/config/derive.py

```diff
@@ -42,33 +42,115 @@ def get_run_duration(config):
     )
 
 
-def get_current_date(config):
-    """Return current_date from configuration dictionary. This function may read from
+def _get_date(config, coupler_res_date_getter):
+    """Return date from configuration dictionary.  This function may read from
     the remote initial_conditions path in the given configuration dictionary.
```

Comment:
> This function may read from
    the remote initial_conditions path in the given configuration dictionary.
 
Maybe doing I/O is pretty confusing...can you pull the I/O up the call-stack?